packages/dds/test-dds-utils/src/ddsFuzzHarness.ts

```diff
@@ -662,17 +682,43 @@ export function mixinAttach<
 			// However, now that we're transitioning to an attached state, the summarizer client should never have any edits.
 			// Thus we use one of the clients we just loaded as the summarizer client, and keep the client around that we generated the
 			// attach summary from.
-			const summarizerClient = clients[0];
-			clients[0] = state.clients[0];
+			let summarizerClient = clients[0];
+			if (clientA.channel.id === "summarizer") {
+				summarizerClient = clientA;
+			} else {
+				clients[0] = state.clients[0];
+			}
 
 			return {
 				...state,
 				isDetached: false,
 				clients,
 				summarizerClient,
 			};
-		}
+		} else if (operation.type === "rehydrate") {
+			const clientA = state.clients[0];
+			assert.equal(state.clients.length, 1);
+			state.isDetached = true;
```

Comment:
this line is redundant with explicitly setting the property on the state object below. I'd lean toward either mutating the given state object or constructing a new one for return, but not doing both.